proto/dwtools/abase/l1.test/Str.test.s

```diff
@@ -8835,6 +8836,80 @@ ghij
 
 //
 
+function strLinesNearestReport( test )
+{
+  let got, expected
+  test.case = 'charsRange[0] one letter';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 1, 7 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'f';
+  test.identical( got.nearest[ 0 ], expected );
+
+  test.case = 'charsRange[0] a few letters';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 7 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'funct';
+  test.identical( got.nearest[ 0 ], expected );
+
+  test.case = 'charsRange[1] one letter';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 6 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'i';
+  test.identical( got.nearest[ 1 ], expected );
+
+  test.case = 'charsRange[1] a few letters';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'ion';
+  test.identical( got.nearest[ 1 ], expected );
+
+  test.case = 'one line report';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 1 });
+  expected = 'function add( x,y ) { return x + y }';
+  test.identical( got.report, '1 : ' + expected );
+
+  test.case = 'multiline report';
+  got = _.strLinesNearestReport({ src : 'a\nb\nc\ndef\ng\nk', charsRangeLeft : [ 3, 5 ], gray : 1, numberOfLines : 5 });
+  // console.log( got );
+  expected = '3 : a\n4 : b\n5 : c\n6 : def\n7 : g';
+  test.identical( got.report, expected );
+
+  test.case = 'number of lines: out of the range';
+  got = _.strLinesNearestReport({ src : 'test0\nfunction add( x,y ) { return x + y }\ntest2\ntest3\ntest4\n test5', charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 2 });
```

Comment:
Please, use short source strings

```js
'test1\n test2\n test3\n test4 \test5'
```

Also, the better solution is to use variables for long strings:
```js
var src = 'test1\n test2\n test3\n test4 \test5';
```